File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/python/service.py
from models import Dog, Cat, Shelter
from database import get_connection, Transaction


def create_dog(name):
    dog = Dog(name)
    if not dog.validate():
        raise ValueError("invalid dog")
    conn = get_connection()
    conn.execute("INSERT INTO dogs VALUES (?)")
    conn.commit()
    return dog


def create_cat(name):
    cat = Cat(name)
    if not cat.validate():
        raise ValueError("invalid cat")
    conn = get_connection()
    conn.execute("INSERT INTO cats VALUES (?)")
    conn.commit()
    return cat


def transfer_animal(animal, shelter):
    txn = Transaction(get_connection())
    txn.execute("UPDATE animals SET shelter_id = ?")
    shelter.add(animal)
    txn.commit()


def list_animals():
    conn = get_connection()
    return conn.execute("SELECT * FROM animals")

File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/typescript/service.ts
import { Dog, Cat, Shelter } from './models';
import { getConnection, Transaction } from './database';

export function createDog(name: string): Dog {
    const dog = new Dog(name);
    if (!dog.validate()) {
        throw new Error("invalid dog");
    }
    const conn = getConnection();
    conn.execute("INSERT INTO dogs VALUES (?)");
    conn.commit();
    return dog;
}

export function createCat(name: string): Cat {
    const cat = new Cat(name);
    if (!cat.validate()) {
        throw new Error("invalid cat");
    }
    const conn = getConnection();
    conn.execute("INSERT INTO cats VALUES (?)");
    conn.commit();
    return cat;
}

export function transferAnimal(animal: any, shelter: Shelter): void {
    const txn = new Transaction(getConnection());
    txn.execute("UPDATE animals SET shelter_id = ?");
    shelter.add(animal);
    txn.commit();
}

export function listAnimals(): any[] {
    const conn = getConnection();
    return conn.execute("SELECT * FROM animals");
}
